src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java

```diff
@@ -157,21 +164,28 @@ public Void run() {
                         // different default for server and proxy
                         keepAliveTimeout = http.getUsingProxy() ? 60 : 5;
                     }
+                } else if (keepAliveTimeout == -2) {
+                    keepAliveTimeout = 0;
                 }
                 // at this point keepAliveTimeout is the number of seconds to keep
                 // alive, which could be 0, if the user specified 0 for the property
                 assert keepAliveTimeout >= 0;
                 if (keepAliveTimeout == 0) {
-                    http.closeServer();
+                    oldClient = http;
                 } else {
                     v = new ClientVector(keepAliveTimeout * 1000);
                     v.put(http);
                     super.put(key, v);
                 }
         } else {
-            v.put(http);
+            oldClient = v.put(http);
         }
```

Comment:
Please correct the indentation here. All code inside the `synchronized(this)` needs to be moved 4-spaces to the right.

Reply:
addressed